testing/trino-tests/src/test/java/io/trino/execution/TestQueryUser.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.units.Duration;
+import io.trino.Session;
+import io.trino.client.ClientSession;
+import io.trino.client.QueryError;
+import io.trino.client.StatementClient;
+import io.trino.plugin.base.security.FileBasedSystemAccessControl;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.testing.DistributedQueryRunner;
+import okhttp3.OkHttpClient;
+import org.testng.annotations.Test;
+
+import java.time.ZoneId;
+import java.util.Locale;
+import java.util.Optional;
+
+import static com.google.common.io.Resources.getResource;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.client.StatementClientFactory.newStatementClient;
+import static io.trino.plugin.base.security.FileBasedAccessControlConfig.SECURITY_CONFIG_FILE;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+public class TestQueryUser
```

Comment:
What does this class do, actually?

Reply:
The main thing I thought this test added was that it makes sure the impersonated user is the one used to check against access control policies. The other test only checks the current_user value.